File: src/main/java/com/HOIIVUtils/hoi4utils/ioexceptions/NullParameterTypeException.java
package com.HOIIVUtils.hoi4utils.ioexceptions;

public class NullParameterTypeException extends Exception {
	public NullParameterTypeException() {
		super();
	}

	public NullParameterTypeException(String message) {
		super(message);
	}

	public NullParameterTypeException(String message, Throwable cause) {
		super(message, cause);
	}

	public NullParameterTypeException(Throwable cause) {
		super(cause);
	}
}

File: _src_old/main/scala/com/hoi4utils/exceptions/NullParameterTypeException.scala
package com.hoi4utils.exceptions

class NullParameterTypeException(message: String, cause: Throwable) extends Exception(message, cause):

  def this(message: String) =
    this(message, null)

  def this(cause: Throwable) =
    // mirror Java's `super(cause)` semantics: detailMessage = cause.toString
    this(if (cause != null) cause.toString else null, cause)
